google/cloud/spanner_v1/merged_result_set.py

```diff
@@ -37,6 +39,16 @@ def __init__(self, batch_snapshot, partition_id, merged_result_set):
         self._queue: Queue[PartitionExecutorResult] = merged_result_set._queue
 
     def run(self):
+        observability_options = getattr(
+            self._batch_snapshot, "observability_options", {}
+        )
+        with trace_call(
+            "CloudSpanner.PartitionExecutor.run",
+            observability_options=observability_options,
+        ):
+            return self.__run()
+
+    def __run(self):
```

Comment:
I think this will not help since we already have a parent trace `run_partitioned_query` in database.py
You can remove this.

Reply:
It does help because usually partitions are run in concurrent threads.